src/Option/README.md

```diff
@@ -0,0 +1,24 @@
+# Option:
+
+Provides an Option component that is passed as a child to the ListBox component.
+
+## API
+
+### `aria-selected` (boolean; optional):
+The ListBox component will handle the `aria-selected` attribute itself, but the `aria-selected` prop can be used as an override, if necessary.
+
+### `children` (element array; required)
+`children` specifies the list of elements that will be displayed within the Option component.
+
+### `className` (string array; optional)
+`className` specifies Bootstrap class names to apply to the Option component. The default is an empty array. (do I mention the existing classnames here?)
+
+### `tag` (string; optional)
+`tag` is used to specify the element type of the rendered Option component. The default is div. Example alternative include li, Button, etc.
+
+### `onOptionSelect` (function, optional)
+`onOptionSelect` is a function that is called when an Option component is selected, by mouse or by keyboard, to perform some additional action.
```

Comment:
The reason I'm surfacing an `onOptionSelect` prop instead of, say `onClick`, because the behavior in such a prop should happen on selection with both the mouse and the keyboard, so this would cover both cases, instead of the user needing to pass in the same function for onClick and onKeyDown (or worse, different functions).

Furthermore, the underlying implementation of the ListBox will actually have to use onMouseDown and not onClick, and I'd prefer not to surface that oddity (especially since many Paragon components expose onClick) to the consumer.

Reply:
I was expecting this prop to be on the parent element: ListBox since it controls what option is selected by passing `isSelected` to children (I was thinking that ListBox would call the given function with the index of the option). Is there a reason to put this on the Option?

Reply:
One reason I put this on the Option is because the clicking handling is done on the Option. If I pass in the `onOptionSelect` to the parent, then I have no way to know when the Option is clicked in order to invoke that function. I think I explain that in my pull request comment. It's one of the things I'm not sure how to handle.

The other reason is that I feel like it's more "correct". You're selecting an Option, so I would expect that the selection process be defined by the Option. I guess I can see both sides of things. If the selection function does the same thing for every Option, then, yeah, it would be less cumbersome to pass in the function into the ListBox once instead of for each Option. But, in our case, for example, the click function is specific to the Option. 

It's something like

```
assets.map((asset, index) => (<li onMouseDown={() => this.onMouseDown(asset, index)});
```

so each list item has an onClick that gets the actual asset as a parameter. Then that click function is able to call the Redux select action creator with the asset as part of the action data.  I'm not sure how I could pass in a single function to the parent that would be able to have that behavior. Does that make sense? Do you have ideas for an alternative?